lib/finitomata.ex

```diff
@@ -29,9 +29,10 @@ defmodule Finitomata do
             current: Transition.state(),
             payload: payload(),
             timer: non_neg_integer(),
-            history: [Transition.state()]
+            history: [Transition.state()],
+            error: any()
```

Comment:
We produce this error, so we are free to tighten the type as strictly as we want.

Reply:
I'm not sure we will always know the error type. 
It comes from `else` in `with` statement. So, to my understanding, it can be unhandled result from `with`, also it can be `RuntimeError`, `ArgumentError` and etc.

Reply:
Ah, indeed, sorry for the noise, disregard it, please.